tests/rptest/services/utils.py

```diff
@@ -131,82 +132,86 @@ class LogSearch(ABC):
         "Exceptional future ignored",
         "UndefinedBehaviorSanitizer",
         "Aborting on shard",
-        "libc++abi: terminating due to uncaught exception",
+        "terminating due to uncaught exception",
         "oversized allocation",
     ]
 
-    def __init__(self, test_context: TestContext, allow_list, logger: Logger) -> None:
+    def __init__(
+        self,
+        test_context: TestContext,
+        allow_list: CompiledLogAllowList,
+        logger: Logger,
+    ) -> None:
         self._context = test_context
         self.allow_list = allow_list
         self.logger = logger
-        self._raise_on_errors = self._context.globals.get(
+        self._raise_on_errors: bool = self._context.globals.get(
             self.RAISE_ON_ERRORS_KEY, True
         )
 
         # Prepare matching terms
-        self.match_terms = self.DEFAULT_MATCH_TERMS
+        self.match_terms: list[str] = list(self.DEFAULT_MATCH_TERMS)
```

Comment:
before this copy, we were actually modifying `DEFAULT_MATCH_TERMS` (an class variable) every time we invoked the constructor, this this was growing over time with more `^ERROR` terms. Doesn't affect the match and I don't even think it affects the perf since the algos used essentially remove redundant strings as a side-effect.